app/assets/javascripts/discourse/app/components/search-menu.js

```diff
@@ -274,19 +296,17 @@ export default class SearchMenu extends Component {
     this.suggestionKeyword = false;
 
     if (!this.search.activeGlobalSearchTerm) {
-      this.search.noResults = false;
```

Comment:
Setting `noResults=false` bewilders me. Shouldn't it be a `true` value?

I checked the commit history on this line of code, and it was implemented with `false` from the very beginning: https://github.com/discourse/discourse/commit/e9b1d29d8bac50abc5997b552457086f3a66462e#diff-ead7ee4722404fc623b16801d990c01e9029bd5c28c712860b0cfc596ae8edd3R111

Instead, I'd change this value to `true` for desktop and mobile views.
But you, for sure, know more about whether this is a safe change to make.

Reply:
resolving this, since the question is not related to the actual work in this PR

the question is stored in https://dev.discourse.org/t/yuriys-code-observations-and-questions/151759